src/helpers/cosmiconfig.ts

```diff
@@ -39,12 +44,70 @@ export function createCosmiConfigSync(moduleName: string, legacy: boolean) {
   return cosmiconfigSync(moduleName, options);
 }
 
-const loadTypeScript: Loader = (...args) => {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const { TypeScriptLoader } = require('cosmiconfig-typescript-loader');
-  return TypeScriptLoader({ transpileOnly: true })(...args);
+const loadTypeScript: Loader = async (filepath, content) => {
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { TypeScriptLoader } = require('cosmiconfig-typescript-loader');
+    return TypeScriptLoader({ transpileOnly: true })(filepath, content);
+  } catch (err) {
+    if (isRequireESMError(err)) {
+      const hash = createHash('sha256').update(content).digest('base64url');
```

Comment:
Just thinking about this hash, what if the config references another file (like a base config or whatever). Changing only the other file will still hit the cache. This might be an edge case though...

On the other hand, how much do we gain by having a cache? Would it be simpler to run `tsc` every time?

Reply:
Yeah, it's something briefly discussed here: https://tortilla-hq.slack.com/archives/CCVDM3NCD/p1677679657142489

From what I tested, it's a 2-3 seconds difference (I can send some benchmarks if you'd like). 

Another idea is to transpile it for now and revisit caching later if we face problems/if people complain about the speed.

Reply:
> From what I tested, it's a 2-3 seconds (for me. on my computer with many processes running) difference (I can send some benchmarks if you'd like).

No need for benchmarks, was just asking if it's a noticeable difference - it seems to be.

What about adding the `--incremental` flag to tsc [here](https://github.com/beerose/graphql-config/blob/c4a65e1420cbb3ec21c900742307382b331c777d/src/helpers/cosmiconfig.ts#L76)? Basically let TS do the caching. Would it still be slow then for you?

Reply:
Using `--incremental` yields this error:
```
error TS5074: Option '--incremental' can only be specified using tsconfig, emitting to single file or when option '--tsBuildInfoFile' is specified.
```

Reply:
Right, lets keep it like it is for now then.